heartbeat/monitors/active/http/http_test.go

```diff
@@ -367,6 +369,24 @@ func runHTTPSServerCheck(
 		time.Sleep(time.Millisecond * 500)
 	}
 
+	// When connecting through a proxy, the following fields are missing.
+	if _, isProxy := reqExtraConfig["proxy_url"]; isProxy {
+		missing := map[string]interface{}{
+			"http.rtt.response_header.us": time.Duration(0),
+			"http.rtt.content.us":         time.Duration(0),
+			"monitor.ip":                  "127.0.0.1",
+			"tcp.rtt.connect.us":          time.Duration(0),
+			"http.rtt.validate.us":        time.Duration(0),
+			"http.rtt.write_request.us":   time.Duration(0),
+			"tls.rtt.handshake.us":        time.Duration(0),
+		}
```

Comment:
I'm not sure why these fields are missing when `proxy_url` is defined. `monitor.ip` worries me.

Reply:
IIRC we intentionally drop `monitor.id` because the IP we've connected to is masked by the proxy. There may be ways to pull that out of special HTTP headers however. 

I haven't looked at that code in a while TBH, and it's never been a concern from users (maybe till now). IIRC there were issues with the timings being relevant. 

TCP connection time is not really as relevant either as that would measure TCP connection time to the proxy, for instance. If this is an area where you'd like to go deep and add relevant test cases we'd be glad to accept a patch, however, as far as I'm aware there isn't huge demand.